### PR TITLE
Accept a `style` parameter to specify the copied style.

### LINKED ISSUE #46: Copied text has a grey background when pasted into gmail or google docs.
This is pretty much the same issue as [#43](https://github.com/sudodoki/copy-to-clipboard/issues/43) but now the background is grey instead of yellow. Do you know what might cause this and if there is a way to make that not happen?

### PR DESCRIPTION
**Pull request by julienw**
If `style` parameter isn't specified, this forces a text/plain copy to make it easier to work with word processors.
This also changes the example code to exhibit the bug of issue #46 so
that it's easier to test.

Fixes #46

### CODE REVIEW

File: example/index.html [hunk 1]
```diff
@@ -23,7 +26,7 @@ <h2 id="multiline-text">Multiline text copy</h2>
 some multiline text
 for us to copy</textarea>
     <div class="half">
-      <button data-test="init-multiline-text">Click to copy multiline text</button>
```
Comment (julienw) [inline example/index.html]: I thought that `data-test` wasn't used at all so I repurposed this to make it easier to target buttons to add the click event listeners.

File: index.d.ts [hunk 2]
```diff
@@ -2,10 +2,15 @@
...
+  background?: string;
+  text?: string;
+}
+
 interface Options {
   debug?: boolean;
   message?: string;
-  format?: string; // MIME type
+  style?: boolean | StyleOption; // MIME type
 }
```
Comment (julienw) [inline index.d.ts]: I don't know about typescript, I didn't check this is working as expected, so please double check!

File: index.js [hunk 3]
```diff
@@ -65,7 +80,7 @@ function copy(text, options) {
     debug && console.error("unable to copy using execCommand: ", err);
     debug && console.warn("trying IE specific stuff");
     try {
-      window.clipboardData.setData(options.format || "text", text);
+      window.clipboardData.setData("text", text);
```
Comment (julienw) [inline index.js]: I checked the documentation on [microsoft website](https://msdn.microsoft.com/en-us/ie/ms535220(v=vs.80)) and there's no possibility to pass rich text. So I reverted to the previous code here.

File: example/index.js [hunk 4]
```diff
@@ -1 +1 @@
...
...Child(l),a.selectNodeContents(l),c.addRange(a),!document.execCommand("copy"))throw new Error("copy command was unsuccessful");s=!0}catch(e){n&&console.error("unable to copy using execCommand: ",e),n&&console.warn("trying IE specific stuff");try{window.clipboardData.setData(o.format||"text",t),s=!0}catch(e){n&&console.error("unable to copy using clipboardData: ",e),n&&console.error("falling back to prompt"),r=function(e){var t=(/mac os x/i.test(navigator.userAgent)?"⌘":"Ctrl")+"+C";return e.replace(/#{\s*key\s*}/g,t)}("message"in o?o.message:"Copy to clipboard: #{key}, Enter"),window.prompt(r,t)}}finally{c&&("function"==typeof c.removeRange?c.removeRange(a):c.removeAllRanges()),l&&document.body.removeChild(l),e()}return s}},{"toggle-selection":2}],2:[function(e,t,o){t.exports=function(){var t=document.getSelection();if(!t.rangeCount)return function(){};for(var e=document.activeElement,o=[],n=0;n<t.rangeCount;n++)o.push(t.getRangeAt(n));switch(e.tagName.toUpperCase()){case"INPUT":case"TEXTAREA":e.blur();break;default:e=null}return t.removeAllRanges(),function(){"Caret"===t.type&&t.removeAllRanges(),t.rangeCount||o.forEach(function(e){t.addRange(e)}),e&&e.focus()}}},{}]},{},[1])(1)});
```
Comment (julienw) [inline example/index.js]: This is the result from running `yarn pretest`, so that I could test using the example code.

**Review by julienw — COMMENTED** [review]
I tried this code with both Chrome and Firefox using the example page, and this seems to work for me.
I tried pasting to a google drive document with white background, and also with a table with red background to check that both the black-on-white and no-formatting cases were working fine too.

I couldn't try in IE, Edge or Safari, so it would be a good idea to try there too.

Sorry for coming up with this solution a bit late, this was in my head for some time and just slipped out of my attention :/ I hope you'll like it...

**Comment by julienw** [comment]
Also, as a quick note, I don't like the previous solution using a `format` property because I think it's not versatile enough. I don't think that copying text using whatever color the `<body>` has is what users would want...

**Comment by julienw** [comment]
ping @sudodoki, hey, do you think you'll be albe to look at this PR soon?
I believe the previous solution 1/ doesn't provide much value, and 2/ doesn't fix the background issue in Chrome. And that this solution brings a much better value. Tell me what you think!

**Comment by sudodoki** [comment]
@julienw if this change is accepted what should user do if they don't want any style changes, yet want a rich text copied?

**Comment by sudodoki** [comment]
TBH, I think I'm slightly reluctant to merge this as this is a breaking change and 1) I'm still not 100% sold this is a right way to fix this, 2) currently I don't have too much time on my hands to deal with this. 🤷‍♂ Although your enthusiasm about this is inspiring.

**Comment by julienw** [comment]
The current situation is less than ideal because:
1. it's kind of broken in chrome if there's a background color on the body (that's issue #46)
2. specifying a format is broken in old IE (but maybe we don't mind much)
3. it's possible to copy a rich text, but we don't control the style, and the behavior is different depending on the browser, so what's the point? I admit I really don't get it :-)

This 3rd point also answers your question `if this change is accepted what should user do if they don't want any style changes, yet want a rich text copied?`. I don't understand the use case.
I think this would work OK cross-browser only on a webpage that doesn't have its background set. As soon as the page has a background set the behavior will be different in Chrome vs Firefox (and probably others that I haven't tried). I personally think it's not a good thing.

That's why in this PR I added some code to force a white background and a black text color, so that we have some cross-browser consistency.

I guess I could try to keep the code using `format` so that it's not a breaking change anymore, and mark it clearly as "obsolete" so that we could remove it later. I can do that if you want to.

**Pull request closed without merging**